src/scripts/core/models/grid.ts

```diff
@@ -13,62 +19,126 @@ export class Grid extends Container {
+            const secondY = (this.availWidth / (this.gridSize/this.selectedTokens[1]._verticalIndex))
+            this.selectedTokens[0].position = {x: secondX, y: secondY}
+            this.selectedTokens[1].position = {x: firstX, y: firstY}
+
+            //set the tokens' skins
             const firstSkIndex = this.selectedTokens[0].skIndex;
             const secondSkIndex = this.selectedTokens[1].skIndex;
-            this.selectedTokens[0].setToken(secondSkIndex);
-            this.selectedTokens[1].setToken(firstSkIndex);
-            this.selectedTokens = [undefined, undefined];
-            this.selectedTokens[0], this.selectedTokens[1] = undefined;
-            this.selectedTokens[0], this.selectedTokens[1] = undefined;
+            this.selectedTokens[0].setSkin(secondSkIndex);
+            this.selectedTokens[1].setSkin(firstSkIndex);
+
+            //THE TOKENS SWAP POSITIONS FIRST!
+            const tweenCurve = new Curve(
+                [this.selectedTokens[0].x, this.selectedTokens[0].y],
+                [this.selectedTokens[1].x, this.selectedTokens[1].y]
+                );
+            
+            const rotateFirst = tweenCurve.rotate90(this.selectedTokens[0].x, this.selectedTokens[0].y);
+            const rotateSecond = tweenCurve.rotate90(this.selectedTokens[1].x, this.selectedTokens[1].y);
+
+            //tween the tokens from their DESTINATION to their ORIGIN
+            const swapTween = gsap.timeline( {
+                ease: "back",
+                onStart: function(): void {
+                    eventEmitter.emit('onSwapStart');
+                },
+                onComplete: function(): void {
+                    eventEmitter.emit('onSwapComplete');
+                }
+            } );
+            
+            swapTween.to(this.selectedTokens[0],
+                {
+                    motionPath: {
+                        curviness: 2,
+                        path: [
+                            {x: rotateFirst[0], y: rotateFirst[1]},
+                            {x: firstX, y: firstY}
+                        ]
+                    },
+                    duration: 1.5
+                }, 0)
+
+            swapTween.to(this.selectedTokens[1],
+                {
+                    motionPath: {
+                        curviness: 2,
+                        path: [
+                            {x: rotateSecond[0], y: rotateSecond[1]},
+                            {x: secondX, y: secondY}
+                        ]
+                    },
+                    duration: 1.5
+                }, 0)
 
-            this.resolveMatches();
             return;
         }
+    }
 
+    private onSwapStart(): void {
+        this.columns.forEach( column => {column.deactivateAllTokens()});
+    }
+
+    private onSwapComplete(): void {
+        this.selectedTokens = [undefined, undefined];
+        this.resolveMatches();
+        this.columns.forEach( column => {column.activateUnMatchedTokens()});
```

Comment:
you might be able to replace this line with this.activateUnMatchedTokens(), would mean any future changes to activating tokens would only need to be changed in one place